runtime/executor/src/main/java/org/apache/nemo/runtime/executor/data/partition/SerializedPartition.java

```diff
@@ -45,11 +48,13 @@
   private volatile boolean committed;
   // Will be null when the partition is committed when it is constructed.
   @Nullable
-  private final DirectByteArrayOutputStream bytesOutputStream;
+  private final DirectByteBufferOutputStream bytesOutputStream;
```

Comment:
I have a feeling that this class can be refactored into multiple classes, without null fields, but that's probably out of the scope of this PR. Can you file a JIRA issue to refactor this class, and add a TODO comment?

Reference for the TODO comment: https://github.com/apache/incubator-nemo/blob/master/runtime/master/src/main/java/org/apache/nemo/runtime/master/scheduler/BatchScheduler.java#L567